### PR TITLE
rsmeter: add centralized place to construct protobuf data

### PR DESCRIPTION
**Pull request by zhongzc**
Signed-off-by: Zhenchi <zhongzc_arch@outlook.com>

### What is changed and how it works?

ref #11691 

What's Changed:

Many datasinks will be interested in report data. It's wasted to convert them from the original shape to the protobuf shape multiple times.

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by mornyx** [comment]
LGTM

**Review by breeswish — APPROVED** [review]

**Review by andylokandy — APPROVED** [review]

File: components/resource_metering/src/reporter/data_sink.rs [hunk 1]
```diff
@@ -10,5 +13,5 @@ pub trait DataSink {
     // by the sink. A deadline can be specified to control how late it should be sent.
     // If the sink is kept full and cannot schedule a send within the specified deadline,
     // or the sink is closed, an error will be returned.
-    fn try_send(&mut self, records: Records) -> Result<()>;
+    fn try_send(&mut self, records: Arc<Vec<ResourceUsageRecord>>) -> Result<()>;
```
[the remark this answers is not in the document]
Comment (mornyx) [reply components/resource_metering/src/reporter/data_sink.rs:16]: This PR has been merged, just a suggestion forgotten to submit :(

[the remark this answers is not in the document]
Comment (zhongzc) [reply components/resource_metering/src/reporter/data_sink.rs:16]: Once an implementation of `DataSink` wants to own a list of `ResourceUsageRecord`, `&[ResourceUsageRecord]` will cause a full clone.

**Pull request merged**